src/clib/pioc.c

```diff
@@ -55,26 +55,66 @@ int PIOc_File_is_Open(int ncid)
 }
 
 /**
- * Set the error handling method to be used for subsequent pio
- * library calls, returns the previous method setting.
+ * Set the error handling method to be used for subsequent pio library
+ * calls, returns the previous method setting. This function is
+ * supported but deprecated. New code should use
+ * PIOc_set_file_error().  This method has no way to return an error,
+ * so any failure will result in MPI_Abort.
  *
  * @param ncid the ncid of an open file
  * @param method the error handling method
- * @returns 0 on success, error code otherwise
+ * @returns old error handler
+ * @ingroup PIO_error_method
  */
 int PIOc_Set_File_Error_Handling(int ncid, int method)
 {
     file_desc_t *file;
     int oldmethod;
     int ret;
 
+    /* Get the file info. */
+    if (pio_get_file(ncid, &file))
+        piodie("Could not find file", __FILE__, __LINE__);                
+
+    /* Get the old method. */
+    oldmethod = file->iosystem->error_handler;
+
+    /* Set the file error handler. */
+    if (PIOc_set_file_error(ncid, method))
+        piodie("Could not set the file error hanlder", __FILE__, __LINE__);        
+
+    return oldmethod;
+}
+
+/**
+ * Set the error handling method to be used for subsequent pio
+ * library calls on this file.
+ *
+ * @param ncid the ncid of an open file
+ * @param method the error handling method
+ * @returns 0 for success, error code otherwise.
+ * @ingroup PIO_error_method
+ */
+int PIOc_set_file_error(int ncid, int method)
```

Comment:
It's also useful and used in many users codes to know what the previous method was so that it can be
reset when the user is done using the new method - if it's not in the return code it should be in the arguments.